### PR TITLE
Finalise delete account feature

### LINKED ISSUE #283: Finalise the "delete account feature" by removing the activities created by the user (Firebase Firestore documents and Firebase Realtime Database related chats)

### LINKED ISSUE #284: Review abstraction usages in ChatActivity and EditProfileActivity

### LINKED ISSUE #285: Review and add tests for ChatActivity and ActivityDescriptionActivity

### PR DESCRIPTION
**Pull request by Antho1426**
This pull request is intended to:
- Finalise the "delete account feature" by ensuring the proper deletion of not only the user and his information (from Firebase Firestore (information), Firebase Storage (profile picture) and Firebase Authentication (his actual Firebase authentication based on his credentials)), but also the activities he created as an organiser (to avoid bugs with `ActivityDescriptionActivity`) (from Firebase Firestore) and their corresponding chats (from Firebase Realtime Database).
- Ensure a decent test coverage by improving the tests for the `ChatActivity` and `ActivityDescriptionActivity`
- Review the abstraction usage in `ChatActivity` (where the temporary message uploaded to Firebase Realtime Database doesn't yet make use of the abstraction) and `EditProfileActivity` (where the "delete account" feature doesn't yet make use of the abstraction for deleting the user profile picture)

### CODE REVIEW

**Review by bitshrine — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityRegisterToActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+import java.util.Date;
+import java.util.concurrent.CountDownLatch;
+
+import static androidx.test.core.app.ApplicationProvider.getApplicationContext;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+@RunWith(AndroidJUnit4.class)
+public class ActivityDescriptionActivityRegisterToActivityTest {
```
Comment (RoxaneBurri) [inline app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityRegisterToActivityTest.java:39]: What are you testing in this class ?

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityUnregisterFromActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+import java.util.Date;
+import java.util.concurrent.CountDownLatch;
+
+import static androidx.test.core.app.ApplicationProvider.getApplicationContext;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+@RunWith(AndroidJUnit4.class)
+public class ActivityDescriptionActivityUnregisterFromActivityTest {
```
Comment (RoxaneBurri) [inline app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityUnregisterFromActivityTest.java:39]: Same question as before

File: app/src/androidTest/java/com/sdp/movemeet/view/chat/ChatActivityTest.java [hunk 3]
```diff
@@ -50,14 +55,18 @@
...
+    private BackendManager<Message> messageManager;
 
     @Before
     public void signIn() {
         CountDownLatch latch = new CountDownLatch(1);
 
         fAuth = AuthenticationInstanceProvider.getAuthenticationInstance();
 
-        fAuth.signInWithEmailAndPassword("movemeet@gmail.com", "password").addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+        fAuth.signInWithEmailAndPassword(TEST_EMAIL, TEST_PASSWORD).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
```
Comment (RoxaneBurri) [inline app/src/androidTest/java/com/sdp/movemeet/view/chat/ChatActivityTest.java:71]: Nice for use general attributs for email and password

File: app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java [hunk 4]
```diff
@@ -176,10 +181,8 @@ private void setButton() {
         }
     }
 
-    private void getParticipantNames() {
+    private void getParticipantNames(Activity activity) {
```
Comment (RoxaneBurri) [inline app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java:179]: Why do you pass the activity in argument ?

File: app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java [hunk 5]
```diff
@@ -262,26 +265,26 @@ private void createAddressView() {
...
+                userManager.update(FirestoreUserManager.USERS_COLLECTION + ImageHandler.PATH_SEPARATOR + userId, REGISTERED_ACTIVITY_FIELD, activity.getDocumentPath(), UPDATE_FIELD_UNION);
+                // Adding the user ID to the array field "participantId" of the Firebase Firestore
+                // activity document
+                activityManager.update(activity.getDocumentPath(), PARTICIPANT_ID_FIELD, userId, UPDATE_FIELD_UNION).addOnSuccessListener(new OnSuccessListener() {
                     @Override
                     public void onSuccess(Object o) {
                         Log.d(TAG, "Participant registered in Firebase Firestore!");
-                        getParticipantNames();
-                        setButton();
-                        // TODO: (By Victor) here --> get activity from the MainMapFragment and update it!
```
Comment (RoxaneBurri) [inline app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java:276]: The todo is done ?

**Review by RoxaneBurri — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityRegisterToActivityTest.java (hunk 1 above)
Comment (Antho1426) [reply app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityRegisterToActivityTest.java:39]: In this test, I unfortunately don't mock anything, but I am testing the fact that the user is properly registered in an activity. More precisely, this test goes through the process of adding the `userId` String to the `participantId` array field of the activity document in Firebase Firestore. If the test passes, it means that the user was properly registered and added to the list of participants!

File: app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityUnregisterFromActivityTest.java (hunk 2 above)
Comment (Antho1426) [reply app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityUnregisterFromActivityTest.java:39]: Yes this is basically the same principle as the previous test. Here I test that the user is properly unregistered from the activity to which he registered. More precisely, the first part of the test consists of first registering the user to this activity and the second part to unregister him (i.e. remove his `userId` String from the `participantId` array field of the activity document in Firebase Firestore).

File: app/src/androidTest/java/com/sdp/movemeet/view/chat/ChatActivityTest.java (hunk 3 above)
Comment (Antho1426) [reply app/src/androidTest/java/com/sdp/movemeet/view/chat/ChatActivityTest.java:71]: Yes, this is cleaner! If we can do something like that in most of our test files for the final submission, that would be great!

File: app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java (hunk 4 above)
Comment (Antho1426) [reply app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java:179]: Yes, good question! This is a needed change in order to be able to use this method of the `ActivityDescriptionActivity` for its corresponding test class. More precisely, this is to allow the "good flow" of the required parameters (contained in `activity`) through the tested sequence of functions.

File: app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java (hunk 5 above)
Comment (Antho1426) [reply app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java:276]: Yes, it is done! And in fact, he just released an optimised version of the "reloading of the activities" based on a "Listener" no later than today!

**Pull request merged**